esphome/components/grove_i2c_motor/grove_i2c_motor.h

```diff
@@ -0,0 +1,224 @@
+      Parameter
+       chl: MOTOR_CHA or MOTOR_CHB
+      Return
+       Null.
+  *************************************************************/
+  void dc_motor_brake(uint8_t chl);
+
+  /*************************************************************
+      Description
+       Stop the motor slowly.
+      Parameter
+       chl: MOTOR_CHA or MOTOR_CHB
+      Return
+       Null.
+  *************************************************************/
+  void dc_motor_stop(uint8_t chl);
+
+  /*************************************************************
+      Description
+       Drive a stepper.
+      Parameter
+       mode:  4 driver mode: FULL_STEP,WAVE_DRIVE, HALF_STEP, MICRO_STEPPING,
+              for more information: https://en.wikipedia.org/wiki/Stepper_motor#/media/File:Drive.png
+       steps: The number of steps to run, range from -32768 to 32767.
+              When steps = 0, the stepper stops.
+              When steps > 0, the stepper runs clockwise. When steps < 0, the stepper runs anticlockwise.
+       rpm:   Revolutions per minute, the speed of a stepper, range from 1 to 300.
+              Note that high rpm will lead to step lose, so rpm should not be larger than 150.
+      Return
+       Null.
+  *************************************************************/
+  void stepper_run(StepperModeTypeT mode, int16_t steps, uint16_t rpm);
+
+  /*************************************************************
+      Description
+       Stop a stepper.
+      Parameter
+       Null.
+      Return
+       Null.
+  *************************************************************/
+  void stepper_stop();
+
+  // keeps moving(direction same as the last move, default to clockwise)
+  /*************************************************************
+      Description
+       Keep a stepper running.
+      Parameter
+       mode:  4 driver mode: FULL_STEP,WAVE_DRIVE, HALF_STEP, MICRO_STEPPING,
+              for more information: https://en.wikipedia.org/wiki/Stepper_motor#/media/File:Drive.png
+       rpm:   Revolutions per minute, the speed of a stepper, range from 1 to 300.
+              Note that high rpm will lead to step lose, so rpm should not be larger than 150.
+       is_cw: Set the running direction, true for clockwise and false for anti-clockwise.
+      Return
+       Null.
+  *************************************************************/
+  void stepper_keep_run(StepperModeTypeT mode, uint16_t rpm, bool is_cw);
+
+ private:
+  uint8_t buffer_[16];
+};
+
+template<typename... Ts> class GROVETB6612FNGMotorRunAction : public Action<Ts...> {
+ public:
+  GROVETB6612FNGMotorRunAction(GroveMotorDriveTB6612FNG *motor) : motor_(motor) {}
+  TEMPLATABLE_VALUE(uint8_t, chl)
+  TEMPLATABLE_VALUE(uint16_t, speed)
+
+  void play(Ts... x) override {
+    auto chl = this->chl_.value(x...);
+    auto speed = this->speed_.value(x...);
+    this->motor_->dc_motor_run(chl, speed);
+  }
+
+ protected:
+  GroveMotorDriveTB6612FNG *motor_;
+};
```

Comment:
If you also `#include "esphome/core/helpers.h" at the top, each action can be made slightly simpler:
```suggestion
template<typename... Ts> class GROVETB6612FNGMotorRunAction : public Action<Ts...>, public Parented<GroveMotorDriveTB6612FNG> {
 public:
  TEMPLATABLE_VALUE(uint8_t, chl)
  TEMPLATABLE_VALUE(uint16_t, speed)

  void play(Ts... x) override {
    auto chl = this->chl_.value(x...);
    auto speed = this->speed_.value(x...);
    this->parent_->dc_motor_run(chl, speed);
  }
};
```